miso-service/src/main/java/uk/ac/bbsrc/tgac/miso/service/impl/DefaultProjectService.java

```diff
@@ -0,0 +1,319 @@
+  @Autowired
+  private SecurityManager securityManager;
+
+  @Override
+  public void addProjectWatcher(Project project, User watcher) throws IOException {
+    authorizationManager.throwIfNotWritable(project);
+    projectStore.addWatcher(project, watcher);
+  }
+
+  @Override
+  public void deleteProjectOverviewNote(ProjectOverview projectOverview, Long noteId) throws IOException {
+    authorizationManager.throwIfNonAdmin();
+    if (noteId == null || noteId.equals(Note.UNSAVED_ID)) {
+      throw new IllegalArgumentException("Cannot delete an unsaved Note");
+    }
+    ProjectOverview managed = projectStore.getProjectOverviewById(projectOverview.getId());
+    Note deleteNote = null;
+    for (Note note : managed.getNotes()) {
+      if (note.getNoteId().equals(noteId)) {
+        deleteNote = note;
+        break;
+      }
+    }
+    if (deleteNote == null) {
+      throw new IOException("Note " + noteId + " not found for ProjectOverview " + projectOverview.getId());
+    }
+    managed.getNotes().remove(deleteNote);
+    projectStore.saveOverview(managed);
+  }
+
+  @Override
+  public Project getProjectByAlias(String projectAlias) throws IOException {
+    return projectStore.getByAlias(projectAlias);
+  }
+
+  // GETS
+  @Override
+  public Project getProjectById(long projectId) throws IOException {
+    return projectStore.get(projectId);
+  }
+
+  @Override
+  public Map<String, Integer> getProjectColumnSizes() throws IOException {
+    return projectStore.getProjectColumnSizes();
+  }
+
+  @Override
+  public ProjectOverview getProjectOverviewById(long overviewId) throws IOException {
+    return projectStore.getProjectOverviewById(overviewId);
+  }
+
+  @Override
+  public Collection<ProjectOverview> listAllOverviewsByProjectId(long projectId) throws IOException {
+    return authorizationManager.filterUnreadable(projectStore.listOverviewsByProjectId(projectId), ProjectOverview::getProject);
+  }
+
+  @Override
+  public Collection<Project> listAllProjects() throws IOException {
+    return authorizationManager.filterUnreadable(projectStore.listAll());
+  }
+
+  @Override
+  public Collection<Project> listAllProjectsBySearch(String query) throws IOException {
+    return authorizationManager.filterUnreadable(projectStore.listBySearch(query));
+  }
+
+  @Override
+  public Collection<Project> listAllProjectsWithLimit(long limit) throws IOException {
+    return authorizationManager.filterUnreadable(projectStore.listAllWithLimit(limit));
+  }
+
+  private Collection<Group> loadManagedGroups(Collection<Group> original) throws IOException {
+    if (original == null)
+      return null;
+    List<Group> managed = new ArrayList<>();
+    for (Group item : original) {
+      managed.add(securityManager.getGroupById(item.getGroupId()));
+    }
+    return managed;
+  }
+
+  // SAVES
+
+  private Collection<User> loadManagedUsers(Collection<User> original) throws IOException {
+    if (original == null)
+      return null;
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Return an empty collection instead of null. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1168)